### PR TITLE
Project Labyrinth - Bäm, Nadia and Therése

### PR DESCRIPTION
**Pull request by BamStugendal**
Link to my deployed project: 
https://blissful-noether-c5e8fc.netlify.app/

My reflections on how this week's project turned out: Good despite technical issues and power outage.

Things I'd like to have clarified or explained in more detail:

### CODE REVIEW

File: code/src/component/GameContainer.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
+import React, { useState, useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { Typewriter } from 'react-typewriting-effect';
```
Comment (LillanEriksson) [inline code/src/component/GameContainer.js:3]: So cool effect! Did not know that it was a package

File: code/src/component/GameContainer.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        // Runs the fetch function generateGameStart from reusable
+        dispatch(generateGameStart());
+    }
+
+    // onGameGenerate renders when username is updated
+    useEffect(onGameGenerate, [username])
+
+    const onGameMoveUpdate = (action) => {
+        setActionDescription(action.description)
+        onTimer();
```
Comment (LillanEriksson) [inline code/src/component/GameContainer.js:36]: It as really nice to have the description as loader, maybe you could have had the timer set for a second or two longer, when the description was long it was a bit stressful to keep up reading

File: code/src/reducers/reusable.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        })
+                .then(res => res.json())
+                .then(data => {
+                    dispatch(game.actions.generateGame(data));
+                    dispatch(game.actions.setFetching(false));
+                })
+    };
+};
+
+
```
Comment (LillanEriksson) [inline code/src/reducers/reusable.js:40]: Nice with the fetches in a separate file!

File: code/src/styling/typewriter.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import styled from 'styled-components';
+    
+export const Wrapper = styled.div`
+    margin: 80px 0 0 20px;
+    padding: 10px;
+    font-size: 20px;
+    line-height: 1.6;
+    color: #000;
+`
+
```
Comment (LillanEriksson) [inline code/src/styling/typewriter.js:10]: Great  with the styling in separate files for each page

File: code/src/reducers/game.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    reducers: {
+        generateGame: (store, action) => {
+            //Update direction, description, coordinates after fetch.
+            store.direction = action.payload.actions;   
+			store.description = action.payload.description;
+            store.coordinates = action.payload.coordinates;
+            
+            //Saves direction, description, coordinates after fetch in history.
+            store.history = [...store.history, action.payload]; 
+        },
```
Comment (LillanEriksson) [inline code/src/reducers/game.js:23]: Does this history-thingy do anything? We could not se a back-button or something that uses this?

File: code/src/reducers/game.js [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+        coordinates: '',
+        history: [],
+        fetching: false,
+        timer: false
+    },
+    reducers: {
+        generateGame: (store, action) => {
+            //Update direction, description, coordinates after fetch.
+            store.direction = action.payload.actions;   
+			store.description = action.payload.description;
```
Comment (LillanEriksson) [inline code/src/reducers/game.js:18]: Indentation is a bit off here on line 18

File: README.md [hunk 7]
```diff
@@ -1,13 +1,14 @@
...
-
-Start by briefly describing the assignment in a sentence or two. Keep it short and to the point.
+Create a project which allows a user to navigate a maze, in the form of a text-based adventure, provided by a backend api.
 
 ## The problem
 
 Describe how you approached to problem, and what tools and techniques you used to solve it. How did you plan? What technologies did you use? If you had more time, what would be next?
 
+ Started by planning the project structure with Figma. Knowing the game road-map posted by a fellow student made us gain time. We used Postman to see what data we were receiving and what to post, in order to build the fetch. We had an idea of how we wanted the game look, googling other text based games and proceeded to the styling.
+ Tehcnlogies used: Figma, Postman, VS Code, Npm,
```
Comment (LillanEriksson) [inline README.md:9]: We think that the use of Postman and Figma was a great choice

**Review by LillanEriksson — APPROVED** [review]
Easy to read with all your comments. Styling looks great both in desktop and mobile. You have succeded with blue level requirements, and we see that you have started with some red level like the history. 
Superimpressed by the typewriter-effect!

File: code/src/component/GameContainer.js [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+    // Set timer to true when funtions start and after 7 sec sets it to false.
+    const onTimer = () => {
+        dispatch(game.actions.setTimer(true))
+        setTimeout(() => {
+            dispatch(game.actions.setTimer(false))
+        }, 7000);
+    }
+
+    //If the fetch or timer is still going, render the choosen action description.
+    if(fetching || timing) {
```
Comment (puj) [inline code/src/component/GameContainer.js:51]: Nice logic!
```javascript
if(isFetching || isTimerActive) {
```

Could be one alternative naming

File: code/src/component/GameStartContainer.js [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+import { Button } from './Button';
+import { StartGameScreen, StartGameCard, StartText, Input } from '../styling/startscreen';
+
+export const GameStartContainer = () => {
+    const dispatch = useDispatch();
+    const [inputValue, setInputValue] = useState('');
+
+    // Update store with given username
+    const setUserName = () => {
+        dispatch(game.actions.updateUserName(inputValue));
```
Comment (puj) [inline code/src/component/GameStartContainer.js:14]: Nothing wrong with 
`dispatch(game.actions.setUsername(inputValue));`

if you were inclined to name the action like this

**Review by puj — APPROVED** [review]
This was a really novel approach.  Showing the action description after clicking it.  I really enjoyed that.  The retro look was really solid as well.  

Good, clean code. Easy to follow and showed off some advanced concepts.  Well done!

**Pull request closed without merging**